pkg/http/http.go

```diff
@@ -113,28 +124,61 @@ func (s *Server) WithTracing(provider trace.TracerProvider) *Server {
 	return s
 }
 
-// WithPrometheus adds a /metrics endpoint and instrument subsequently enabled groups with general http-level metrics.
-func (s *Server) WithPrometheus() *Server {
-	// Add MW with .With instead of .Use, as .Use does not allow registering MWs after routes.
-	s.router = s.router.With(PrometheusMiddleware)
-	s.router.Handle("/metrics", promhttp.Handler())
-
+// WithInsecureTraceContext configures the server to trust parent trace IDs arriving to public-facing endpoints.
+// This will make traces generated by QuickPizza to be children of client-controlled trace IDs.
+func (s *Server) WithInsecureTraceContext() *Server {
+	s.insecureTracing = true
```

Comment:
nit: We could add an explicit `false` to the `Server` constructor function.

Reply:
I'm not sure this adds more clarity than noise, I think go developers are used to false-when-omitted booleans 🤔